[ERROR] Sandboxing is not working on your platform neon:

[ERROR] Sandboxing is not working on your platform neon:
        "~/.opam/opam-init/hooks/sandbox.sh build sh -c echo SUCCESS | tee check-write" exited
        with code 1 "bwrap: loopback: Failed RTM_NEWADDR: Operation not permitted"
Do you want to disable it?  Note that this will result in less secure package builds, so please
ensure that you have some other isolation mechanisms in place (such as running within a
container or virtual machine). [y/N]

--- a/master_changes.md
+++ b/master_changes.md
@@ -99,6 +99,7 @@ users)
   * Use variables instead of plain paths [#6760 @rjbou]
   * Reword apparmor message when user need to check the profile [#6760 @rjbou]
   * Add `2.5.0` to the installers [#6817 @kit-ty-kate]
+  * Fix the AppArmor support when installing in `/usr/bin` [#6823 @kit-ty-kate - fix #6820]
 
 ## Admin
 

--- a/shell/install.sh
+++ b/shell/install.sh
@@ -808,6 +808,7 @@ echo "## opam $VERSION installed to $BINDIR"
 if [ -f /etc/apparmor.d/abi/4.0 ] && [ "$(aa-enabled 2> /dev/null)" = Yes ]; then
   TMP_APPARMOR_PROFILE=/tmp/opam-local.aa.tmp
   APPARMOR_PROFILE=/etc/apparmor.d/opam-local
+  UPSTREAM_APPARMOR_PROFILE=/etc/apparmor.d/opam
   cat << EOF > "$TMP_APPARMOR_PROFILE"
 # This profile allows everything and only exists to give the
 # application a name instead of having the label "unconfined"
@@ -826,7 +827,10 @@ EOF
   SKIP_APPARMOR=0
   APPARMOR_CREATION_OPTION="-a"
   if [ -e "$APPARMOR_PROFILE" ]; then
-    if diff -q "$TMP_APPARMOR_PROFILE" "$APPARMOR_PROFILE"; then
+    if [ -f "$UPSTREAM_APPARMOR_PROFILE" ] && [ "$(realpath "$BINDIR")" = /usr/bin ]; then
+      APPARMOR_PROFILE=$UPSTREAM_APPARMOR_PROFILE
+      SKIP_APPARMOR=1
+    elif diff -q "$TMP_APPARMOR_PROFILE" "$APPARMOR_PROFILE"; then
       SKIP_APPARMOR=1
     else
       echo "## The opam-local AppArmor profile already exists and differs from the expected content."

